### PR TITLE
Optimize build for readme changes only

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/25382

## Description
For readme changes only, not creating the servers for tests.
The build now will take about 8-9 minutes.

## Screenshots
Paste here any images that will help the reviewer
image

### CODE REVIEW

**Comment by Shellyber** [comment]
Example of commit with only README change: https://app.circleci.com/pipelines/github/demisto/content/47894/workflows/4624d214-e422-4967-8d33-6b7e4235bcde/jobs/205043

Example of regular commit, changed also a py file:
https://app.circleci.com/pipelines/github/demisto/content/47804/workflows/722367c0-29a2-4385-875d-8fad5f0ab057/jobs/204830

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -1042,7 +1042,13 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
 
     from_version, to_version = get_from_version_and_to_version_bounderies(all_modified_files_paths, id_set)
 
-    create_filter_envs_file(from_version, to_version)
+    # Check if only README file in file string, if so, no need to create the servers.
+    files = [s for s in files_string.split('\n') if s]
+    readme = all(map(lambda s: 'README.md' in s, files))
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1047]: note that we should also handle the case only images/videos were modified/added

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 2]
```diff
@@ -1042,7 +1042,13 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
...
+    if readme:
+        create_filter_envs_file(from_version, to_version, readme_only=True)
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1049]: no need to call `create_filter_envs_file` twice, just build the args dict and pass them to the func

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 3]
```diff
@@ -1156,7 +1162,8 @@ def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id
     return min_from_version.vstring, max_to_version.vstring
 
 
-def create_filter_envs_file(from_version: str, to_version: str, two_before_ga=None, one_before_ga=None, ga=None):
+def create_filter_envs_file(from_version: str, to_version: str, two_before_ga=None, one_before_ga=None, ga=None,
```
Comment (Itay4) [inline Tests/scripts/collect_tests_and_content_packs.py:1165]: please make sure all args have type hints and are in the docstring

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 1 above)
Comment (Shellyber) [reply Tests/scripts/collect_tests_and_content_packs.py:1047]: Images and videos are included only in the readme file, no?
What is the difference?

File: .circleci/config.yml [hunk 4]
```diff
@@ -875,6 +875,13 @@ jobs:
                 when: always
```
Comment (hod-alpert) [inline .circleci/config.yml:875]: Rename this step in both 6.0 and master to `Collect Instance Role And Check If Instance Was Created`

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 1 above)
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1047]:
```suggestion
    documentation_changes_only = all(map(lambda s: s.endswith('.md') or s.endswith('.png') or s.endswith('.jpg') or s.endswith('.mp4'), files))
```

We have also Description.md files.

@Itay4 Does this cover all the cases?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 5]
```diff
@@ -1156,7 +1162,8 @@ def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id
...
+                            readme_only=False):
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1166]:
```suggestion
                            documentation_changes_only=False):
```

**Review by hod-alpert — CHANGES_REQUESTED** [review]
Added some comments
In additions:
Make a function that takes the files string as argument and return True or False as needed and then add unittests for that function.

I naddition - once you have that - run a build where this function returns True and let's see how such build looks like.

**Comment by Shellyber** [comment]
Adding the builds after all the changes:
Only README file: https://app.circleci.com/pipelines/github/demisto/content/48268/workflows/b0ff6219-5144-4879-9b75-ac0e1051f581/jobs/206601

Build with also py file changes: https://app.circleci.com/pipelines/github/demisto/content/48273/workflows/06058670-4456-4590-8b24-433e9ae3ba6f/jobs/206624

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 6]
```diff
@@ -1041,7 +1060,9 @@ def get_test_list_and_content_packs_to_install(files_string, branch_name, minimu
 
     from_version, to_version = get_from_version_and_to_version_bounderies(all_modified_files_paths, id_set)
 
-    create_filter_envs_file(from_version, to_version)
+    # Check if only README file in file string, if so, no need to create the servers.
+    documentation_changes_only = is_documentation_changes_only(files_string)
+    create_filter_envs_file(from_version, to_version, documentation_changes_only=bool(documentation_changes_only))
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1065]:
```suggestion
    create_filter_envs_file(from_version, to_version, documentation_changes_only=documentation_changes_only)
```
Why is the casting required?

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 7]
```diff
@@ -1155,8 +1176,19 @@ def get_from_version_and_to_version_bounderies(all_modified_files_paths: set, id
...
+                            ga: str = None, documentation_changes_only: bool = False):
+    """
+    Create a file containing all the envs we need to run for the CI
+    Args:
+        from_version: Server from_version
+        to_version: Server to_version
+        two_before_ga: Server version two_before_ga
+        one_before_ga: Server version one_before_ga (5.0)
+        ga: Server Version ga (6.0)
+        documentation_changes_only: If the build is for readme_only - no need to create instances.
```
Comment (hod-alpert) [inline Tests/scripts/collect_tests_and_content_packs.py:1189]:
```suggestion
        documentation_changes_only: If the build is for documentations changes only- no need to create instances.
```

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 8]
```diff
@@ -1196,3 +1196,13 @@ def test_remove_tests_for_non_supported_packs(tests_to_filter, should_test_conte
         logging.debug.assert_called_once_with(
             'The following test playbooks are not supported and will not be tested: \n{} '.format(
                 '\n'.join(filtered_tests)))
+
+
+@pytest.mark.parametrize('files_string, expected_result', [
```
Comment (hod-alpert) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:1201]: Add a case with each of the files ending - jpg, mp4 and png to make sure the test will catch any change of behavior

**Review by hod-alpert — APPROVED** [review]
Left some comments
Other then that looks good

**Pull request merged**